### DISCUSSION TITLE
Popover RAC using Framer Motion?

### DISCUSSION THREAD

**Discussion by binaryartifex** [adobe/react-spectrum#5947] [Q&A] [answered]
This is also in reference to a discussion i started that kinda hit a dead end at #5324 concerning an alternative animation solution to the Popover component using framer motion instead of the exceedingly outdated and rarely maintained alternative, tailwindcss-animate. I got as far as the following, if i remove the predicate below the AnimatePresence, it works (without animation obviously) and it console logs the opening and closing of the component. However when i include the predicate even the state context itself no longer registers the popover opening or closing...none of the examples provided on the website show a popover being animated via framer motion.

```typescript
import { AnimatePresence, motion } from "framer-motion";
import { useContext, useEffect } from "react";
import { Popover as AriaPopover, OverlayTriggerStateContext } from "react-aria-components";
import { popoverStyles } from "./popover.styles";
import type { PopoverProps } from "./popover.types";

const MotionPopover = motion(AriaPopover);

export function Popover({ children, ...props }: PopoverProps) {
  const state = useContext(OverlayTriggerStateContext);
  const styles = popoverStyles();

  useEffect(() => {
    console.log(state?.isOpen);
  }, [state]);

  return (
    <AnimatePresence>
      {state?.isOpen && (
        <MotionPopover
          {...props}
          animate={{ opacity: 1 }}
          initial={{ opacity: 0 }}
          exit={{ opacity: 0 }}
          className={styles}
          isOpen
          onOpenChange={state?.setOpen}
        >
          {children}
        </MotionPopover>
      )}
    </AnimatePresence>
  );
}

export default Popover;
```

**Comment by reidbarber** [comment]
Hi @binaryartifex, sorry for the delayed response. It isn't initially clear to me why this isn't working as expected, but if this is still an issue you're looking to solve, feel free to get a demo running in stackblitz/codesandbox, and we can dig deeper!

**Reply by binaryartifex** [reply]
@reidbarber gday mate finally got around to whipping up a rudimentary sandbox that illustrates the issue. would be bloody awesome if someone could take a geez at this when they get a minute. im out of ideas. i just can't figure out why the state suddenly stops propagating to the underlying popover...

[demo sandbox](https://codesandbox.io/p/sandbox/gallant-marco-jdqncg?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clvoglbok0006356ilfhjrooc%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clvoglbok0002356imayh9uic%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clvoglbok0003356i4acdamnu%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clvoglbok0005356iwf6vpyyn%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clvoglbok0002356imayh9uic%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvoglboj0001356i1yeqhtuc%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clvoglbok0002356imayh9uic%2522%252C%2522activeTabId%2522%253A%2522clvoglboj0001356i1yeqhtuc%2522%257D%252C%2522clvoglbok0005356iwf6vpyyn%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvoglbok0004356iwtnkfjet%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clvoglbok0005356iwf6vpyyn%2522%252C%2522activeTabId%2522%253A%2522clvoglbok0004356iwtnkfjet%2522%257D%252C%2522clvoglbok0003356i4acdamnu%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clvoglbok0003356i4acdamnu%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by binaryartifex** [comment]
@reidbarber sorry for the poke mate but i dont spose there's been any movement on this front? kind of a buzzkill atm not having a nice framer motion powered popover transition. its the underlying issue that somehow, AnimatePresence is preventing context state from either entering or leaving the popover, which might suggest a deeper issue....breaking up the modal into modal and modaloverlay works a treat by wrapping them the exact same way...

**Reply by reidbarber** [reply]
Thanks for the sandbox!

Just took a look and I don't think it's related to AnimatePresence, because I can remove it and still see the issue when conditionally rendering the Popover. I'm not sure why it would matter for the trigger state if the Popover has been mounted yet or not. Even when throwing in two Popovers, one conditionally rendered and one not, neither render.

I think it may be worth creating an issue to get a deeper investigation.

**Reply by binaryartifex** [reply, marked as answer]
done mate. issue is at #6369. Closing this discussion in favour of the issue, cheers for the help so far. 👍
